Runtime/C#/Runtime/BebopRecord.cs

```diff
@@ -9,6 +10,36 @@
 
 namespace Bebop.Runtime
 {
+    public abstract class BaseBebopRecord
+    {
+        protected BaseBebopRecord() { }
+
+        /// <summary>
+        /// Gets the maximum number of bytes that will be produced when the current record is encoded.
+        /// </summary>
+        /// <remarks>
+        /// <para>To get the exact amount of bytes that will be produced use <see cref="ByteCount"/>.</para>
+        /// <para>To get the maximum array size, you use <see cref="MaxByteCount"/>. 
```

Comment:
I think it is pretty important to document what "maximum array size" here means. I would phrase it like:

> `MaxByteCount` returns an upper bound on the amount of bytes that will be produced when this record is encoded. It makes the worst-case assumption that every character in every string maps to 4 bytes of UTF-8. That is: to measure each string in the record, it calls `UTF8Encoding.GetMaxByteCount` instead of `UTF8Encoding.GetByteCount`.
> 
> This executes faster than `ByteCount`, but it can return large values that, when used to allocate a buffer, lead to wasted memory. You may prefer this method if your packet contains just a few small strings; you may want to avoid it if your packet contains any very large strings.